### ISSUE TITLE
Taking several picture leads to OutOfMemory Exeption

### ISSUE THREAD

**Issue by tsonevn** [NativeScript/NativeScript#2460]
In case you are using [CameraModule](https://docs.nativescript.org/hardware/camera) and you take several pictures one after another, the application will throw an exception `OutOfMemory Error`. This problem has been reproducible on Android device. 
Adding specific `width` and `height` for the image as arguments in `takePicture` method will not solve this issue. 
I am attaching a sample project, where this problem could be reproduced and debugged locally. 
[https://github.com/tsonevn/SampleCameraModule.git](https://github.com/tsonevn/SampleCameraModule.git)

tns --version 2.1.1
tns-core-modules: 2.1.0
tns-android: 2.1.1

**Comment by toniclecb** [comment]
In my case, this `OutOfMemoryError` is caused by a variable `bitmap` in camera.android.js.
I solved this problem modifying camera.android.js:
```
if (reqHeight_1 > 0 && reqWidth_1 > 0) {
    if (shouldKeepAspectRatio_1) {
        var common = require("./camera-common");
        var aspectSafeSize = common.getAspectSafeDimensions(bitmap.getWidth(), bitmap.getHeight(), reqWidth_1, reqHeight_1);
        scaledSizeImage = android.graphics.Bitmap.createScaledBitmap(bitmap, aspectSafeSize.width, aspectSafeSize.height, true);
    }
    else {
        scaledSizeImage = android.graphics.Bitmap.createScaledBitmap(bitmap, reqWidth_1, reqHeight_1, true);
    }
    bitmap.recycle(); // added line
    bitmap = null; // added line
}
else {
    scaledSizeImage = bitmap;
}
```

**Comment by tsonevn** [comment]
Hi @toniclecb, 

I have tested this solution once again in sample project on android device and notice that this would  temporarily solve the problem.  The memory is still increasing on every taken picture and after you take  10  picture, the app will crash with OutOfMemory Exception. However the time when the app will crash depends on the set width and height, which have been set to `takePickture` method.

Regards,
@tsonevn

**Comment by toniclecb** [comment]
Hi, @tsonevn,
The solution I posted here works when is used takePicture() with width and heigth.
But, in both cases (takePicture with or without params) I need to do:
```
    picture.recycle();
    picture = null;
```
with the returned variable from takePicture().

**Comment by Daxito** [comment]
If after all those lines to clean the memory you call 
**gc()**

Then it works everytime!!!

**Comment by ComputerTinker** [comment]
I am struggling with this issue as well, and it's quite badly noticeable. On a Galaxy S3 running Android 4.4.2, running an app created using NativeScript 2.2.0, if I run takePicture() without passing it any options, I get an out of memory error on the second photo I try to take.  If I run takePicture with { width: 600, height: 600, keepAspectRatio: true }, then my app blows up on the 6th photo.

I tried running the recycle() and gc() functions after I was done referencing the returned camera picture, but then my app was complaining about reusing a recycled image. Hmmm.

**Comment by tsonevn** [comment]
This issue has been fixed with NativeScript 2.3.  Could you verify, whether your problem has been fixed after [upgrade](https://docs.nativescript.org/releases/upgrade-instructions).

**Comment by ComputerTinker** [comment]
I appreciate the fix, however, I had to switch away from Nativescript to a different framework in order to try to finish my project on time, so I will be unable to test for the time being. (Also, the Telerik Platform that I was using only allowed certain versions of Nativescript to be selected, and was still maxed out at 2.2 last time I checked.)
